### ISSUE TITLE
[BUG] Crash with FileReader

### ISSUE THREAD

**Issue by MajorCooke** [ZDoom/gzdoom#2423]
Null pointer crash occurs sporadically when playing the game. Points to fs_files.h.

```
>	[Inline Frame] gzdoom.exe!FileSys::FileReader::Read(void *) Line 266	C++
 	[Inline Frame] gzdoom.exe!FileSys::DecompressorZ::FillBuffer() Line 232	C++
 	[Inline Frame] gzdoom.exe!FileSys::DecompressorZ::Open(FileSys::FileReader *) Line 164	C++
 	gzdoom.exe!FileSys::OpenDecompressor(FileSys::FileReader & self, FileSys::FileReader & parent, __int64 length, int method, int flags) Line 870	C++
 	gzdoom.exe!FileSys::FResourceFile::GetEntryReader(unsigned int entry, int readertype, int readerflags) Line 708	C++
 	gzdoom.exe!FileSys::FileSystem::OpenFileReader(int lump, int readertype, int readerflags) Line 1316	C++
 	[Inline Frame] gzdoom.exe!FileSys::FileSystem::OpenFileReader(int) Line 101	C++
 	gzdoom.exe!FPNGTexture::CopyPixels(FBitmap * bmp, int conversion, int frame) Line 568	C++
 	gzdoom.exe!FImageSource::GetCachedBitmap(const PalEntry * remap, int conversion, int * ptrans, int frame) Line 270	C++
 	gzdoom.exe!FImageTexture::GetBgraBitmap(const PalEntry * p, int * trans) Line 83	C++
 	gzdoom.exe!FTexture::CreateTexBuffer(int translation, int flags) Line 366	C++
 	gzdoom.exe!FTexture::TrimBorders(unsigned short * rect) Line 451	C++
 	gzdoom.exe!FGameTexture::SetupSpriteData() Line 347	C++
 	gzdoom.exe!FGameTexture::GetSpritePositioning(int which) Line 313	C++
 	gzdoom.exe!HWSprite::Process(HWDrawInfo * di, AActor * thing, sector_t * sector, area_t in_area, int thruportal, bool isSpriteShadow) Line 964	C++
 	gzdoom.exe!HWDrawInfo::RenderThings(subsector_t * sub, sector_t * sector) Line 530	C++
 	gzdoom.exe!HWDrawInfo::WorkerThread() Line 115	C++
 	[External Code]	
 	[Inline Frame] gzdoom.exe!std::_Func_class<void,int>::operator()(int <_Args_0>) Line 878	C++
 	gzdoom.exe!ctpl::thread_pool::set_thread::__l2::<lambda_1>::operator()() Line 219	C++
 	[External Code]	
 	gzdoom.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Line 97	C++
 	[External Code]	
```

I was trying to debug a sudden freeze in GZDoom when this happens.

Doing this with RelWithDebInfo build. This happens with whatever mod I try, sometimes even while modless.

**Comment by RicardoLuis0** [comment]
did you check errno? this is most likely due to fopen failing with errno = EMFILE (#2413)

**Comment by MajorCooke** [comment]
How can I do that?

**Comment by coelckers** [comment]
fixed in ef4c4ee503c2faa6a9c0ba2ed32613adfb01ce27

That stack trace was what I really needed to track it down.

**Comment by prof-hastig** [comment]
#2407 and #2408 may also be related, they look like the same kind of unchecked file access.
